File: embroidermodder2/mainwindow-menus.cpp
#include "mainwindow.h"
#include <QDebug>
#include <QMenu>
#include <QMenuBar>
#include <QAction>

void MainWindow::createFileMenu()
{
    qDebug("MainWindow createFileMenu()");
    menuBar()->addMenu(fileMenu);
    fileMenu->addAction(actionHash.value(ACTION_NEW));
    fileMenu->addSeparator();
    fileMenu->addAction(actionHash.value(ACTION_OPEN));

    fileMenu->addMenu(recentMenu);
    connect(recentMenu, SIGNAL(aboutToShow()), this, SLOT(recentMenuAboutToShow()));
    //Do not allow the Recent Menu to be torn off. It's a pain in the ass to maintain.
    recentMenu->setTearOffEnabled(false);

    fileMenu->addSeparator();
    fileMenu->addAction(actionHash.value(ACTION_SAVE));
    fileMenu->addAction(actionHash.value(ACTION_SAVE_AS));
    fileMenu->addSeparator();
    fileMenu->addAction(actionHash.value(ACTION_PRINT));
    fileMenu->addSeparator();
    fileMenu->addAction(actionHash.value(ACTION_WINDOW_CLOSE));
    fileMenu->addSeparator();
    fileMenu->addAction(actionHash.value(ACTION_DESIGN_DETAILS));
    fileMenu->addSeparator();

    fileMenu->addAction(actionHash.value(ACTION_EXIT));
    fileMenu->setTearOffEnabled(false);
}

void MainWindow::createEditMenu()
{
    qDebug("MainWindow createEditMenu()");
    menuBar()->addMenu(editMenu);
    editMenu->addAction(actionHash.value(ACTION_UNDO));
    editMenu->addAction(actionHash.value(ACTION_REDO));
    editMenu->addSeparator();
    editMenu->addAction(actionHash.value(ACTION_CUT));
    editMenu->addAction(actionHash.value(ACTION_COPY));
    editMenu->addAction(actionHash.value(ACTION_PASTE));
    editMenu->addSeparator();
    editMenu->setTearOffEnabled(true);
}

void MainWindow::createViewMenu()
{
    qDebug("MainWindow createViewMenu()");

    QString appDir = qApp->applicationDirPath();
    QString icontheme = getSettingsGeneralIconTheme();

    menuBar()->addMenu(viewMenu);
    viewMenu->addSeparator();
    viewMenu->addMenu(zoomMenu);
    zoomMenu->setIcon(QIcon(appDir + "/icons/" + icontheme + "/zoom" + ".png"));
    zoomMenu->addAction(actionHash.value(ACTION_ZOOM_REAL_TIME));
    zoomMenu->addAction(actionHash.value(ACTION_ZOOM_PREVIOUS));
    zoomMenu->addSeparator();
    zoomMenu->addAction(actionHash.value(ACTION_ZOOM_WINDOW));
    zoomMenu->addAction(actionHash.value(ACTION_ZOOM_DYNAMIC));
    zoomMenu->addAction(actionHash.value(ACTION_ZOOM_SCALE));
    zoomMenu->addAction(actionHash.value(ACTION_ZOOM_CENTER));
    zoomMenu->addSeparator();
    zoomMenu->addAction(actionHash.value(ACTION_ZOOM_IN));
    zoomMenu->addAction(actionHash.value(ACTION_ZOOM_OUT));
    zoomMenu->addSeparator();
    zoomMenu->addAction(actionHash.value(ACTION_ZOOM_SELECTED));
    zoomMenu->addAction(actionHash.value(ACTION_ZOOM_ALL));
    zoomMenu->addAction(actionHash.value(ACTION_ZOOM_EXTENTS));
    viewMenu->addMenu(panMenu);
    panMenu->setIcon(QIcon(appDir + "/icons/" + icontheme + "/pan" + ".png"));
    panMenu->addAction(actionHash.value(ACTION_PAN_REAL_TIME));
    panMenu->addAction(actionHash.value(ACTION_PAN_POINT));
    panMenu->addSeparator();
    panMenu->addAction(actionHash.value(ACTION_PAN_LEFT));
    panMenu->addAction(actionHash.value(ACTION_PAN_RIGHT));
    panMenu->addAction(actionHash.value(ACTION_PAN_UP));
    panMenu->addAction(actionHash.value(ACTION_PAN_DOWN));
    viewMenu->addSeparator();
    viewMenu->addAction(actionHash.value(ACTION_DAY));
    viewMenu->addAction(actionHash.value(ACTION_NIGHT));
    viewMenu->addSeparator();

    viewMenu->setTearOffEnabled(true);
    zoomMenu->setTearOffEnabled(true);
    panMenu->setTearOffEnabled(true);
}

void MainWindow::createSettingsMenu()
{
    qDebug("MainWindow createSettingsMenu()");
    menuBar()->addMenu(settingsMenu);
    settingsMenu->addAction(actionHash.value(ACTION_SETTINGS_DIALOG));
    settingsMenu->addSeparator();
    settingsMenu->setTearOffEnabled(true);
}

void MainWindow::createWindowMenu()
{
    qDebug("MainWindow createWindowMenu()");
    menuBar()->addMenu(windowMenu);
    connect(windowMenu, SIGNAL(aboutToShow()), this, SLOT(windowMenuAboutToShow()));
    //Do not allow the Window Menu to be torn off. It's a pain in the ass to maintain.
    windowMenu->setTearOffEnabled(false);

}

void MainWindow::createHelpMenu()
{
    qDebug("MainWindow createHelpMenu()");
    menuBar()->addMenu(helpMenu);
    helpMenu->addAction(actionHash.value(ACTION_HELP));
    helpMenu->addSeparator();
    helpMenu->addAction(actionHash.value(ACTION_CHANGELOG));
    helpMenu->addSeparator();
    helpMenu->addAction(actionHash.value(ACTION_TIP_OF_THE_DAY));
    helpMenu->addSeparator();
    helpMenu->addAction(actionHash.value(ACTION_ABOUT));
    helpMenu->addSeparator();
    helpMenu->addAction(actionHash.value(ACTION_WHATS_THIS));
    helpMenu->setTearOffEnabled(true);
}

void MainWindow::createAllMenus()
{
    qDebug("MainWindow createAllMenus()");
    createFileMenu();
    createEditMenu();
    createViewMenu();
    createSettingsMenu();
    createWindowMenu();
    createHelpMenu();

}

File: experimental/python/gui/mainwindow_menus.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

"""
====================================
|module_summary| mainwindow_menus.py
====================================

TOWRITE

These methods are to be imported into the MainWindow class directly.

"""

#-Imports.----------------------------------------------------------------------
#--Python Imports.
import os

#--PySide/PyQt Imports.
if PYSIDE:
    from PySide.QtCore import qDebug, SIGNAL, SLOT
    from PySide.QtGui import QIcon
elif PYQT4:
    import sip
    sip.setapi('QString', 2)
    sip.setapi('QVariant', 2)
    from PyQt4.QtCore import qDebug, SIGNAL, SLOT
    from PyQt4.QtGui import QIcon


# C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++
#include "mainwindow.h"
#include <QDebug>
#include <QMenu>
#include <QMenuBar>
#include <QAction>
# C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++C++

# meth MainWindow::
# These methods are to be imported into the MainWindow class.
# Just imagine the methods are right inside it.


def createFileMenu(self):  # void MainWindow::createFileMenu()
    """
    TOWRITE
    """
    actionHash = self.actionHash
    qDebug("MainWindow createFileMenu()")
    self.menuBar().addMenu(self.fileMenu)
    self.fileMenu.addAction(actionHash["ACTION_new"])
    self.fileMenu.addSeparator()
    self.fileMenu.addAction(actionHash["ACTION_open"])

    self.fileMenu.addMenu(self.recentMenu)
    self.connect(self.recentMenu, SIGNAL("aboutToShow()"), self, SLOT("recentMenuAboutToShow()"))
    # Do not allow the Recent Menu to be torn off. It's a pain in the ass to maintain.
    self.recentMenu.setTearOffEnabled(False)

    self.fileMenu.addSeparator()
    self.fileMenu.addAction(actionHash["ACTION_save"])
    self.fileMenu.addAction(actionHash["ACTION_saveas"])
    self.fileMenu.addSeparator()
    self.fileMenu.addAction(actionHash["ACTION_print"])
    self.fileMenu.addSeparator()
    self.fileMenu.addAction(actionHash["ACTION_windowclose"])
    self.fileMenu.addSeparator()
    self.fileMenu.addAction(actionHash["ACTION_designdetails"])
    self.fileMenu.addSeparator()

    self.fileMenu.addAction(actionHash["ACTION_exit"])
    self.fileMenu.setTearOffEnabled(False)

def createEditMenu(self):  # void MainWindow::createEditMenu()
    """
    TOWRITE
    """
    actionHash = self.actionHash
    qDebug("MainWindow createEditMenu()")
    self.menuBar().addMenu(self.editMenu)
    self.editMenu.addAction(actionHash["ACTION_undo"])
    self.editMenu.addAction(actionHash["ACTION_redo"])
    self.editMenu.addSeparator()
    self.editMenu.addAction(actionHash["ACTION_cut"])
    self.editMenu.addAction(actionHash["ACTION_copy"])
    self.editMenu.addAction(actionHash["ACTION_paste"])
    self.editMenu.addSeparator()
    self.editMenu.setTearOffEnabled(True)

def createViewMenu(self):  # void MainWindow::createViewMenu()
    """
    TOWRITE
    """
    actionHash = self.actionHash
    qDebug("MainWindow createViewMenu()")

    icontheme = self.getSettingsGeneralIconTheme()  # QString

    self.menuBar().addMenu(self.viewMenu)
    self.viewMenu.addSeparator()
    self.viewMenu.addMenu(self.zoomMenu)
    self.zoomMenu.setIcon(QIcon(self.gIconDir + os.sep + icontheme + "/zoom" + ".png"))
    self.zoomMenu.addAction(actionHash["ACTION_zoomrealtime"])
    self.zoomMenu.addAction(actionHash["ACTION_zoomprevious"])
    self.zoomMenu.addSeparator()
    self.zoomMenu.addAction(actionHash["ACTION_zoomwindow"])
    self.zoomMenu.addAction(actionHash["ACTION_zoomdynamic"])
    self.zoomMenu.addAction(actionHash["ACTION_zoomscale"])
    self.zoomMenu.addAction(actionHash["ACTION_zoomcenter"])
    self.zoomMenu.addSeparator()
    self.zoomMenu.addAction(actionHash["ACTION_zoomin"])
    self.zoomMenu.addAction(actionHash["ACTION_zoomout"])
    self.zoomMenu.addSeparator()
    self.zoomMenu.addAction(actionHash["ACTION_zoomselected"])
    self.zoomMenu.addAction(actionHash["ACTION_zoomall"])
    self.zoomMenu.addAction(actionHash["ACTION_zoomextents"])
    self.viewMenu.addMenu(self.panMenu)
    self.panMenu.setIcon(QIcon(self.gIconDir + os.sep + icontheme + "/pan" + ".png"))
    self.panMenu.addAction(actionHash["ACTION_panrealtime"])
    self.panMenu.addAction(actionHash["ACTION_panpoint"])
    self.panMenu.addSeparator()
    self.panMenu.addAction(actionHash["ACTION_panleft"])
    self.panMenu.addAction(actionHash["ACTION_panright"])
    self.panMenu.addAction(actionHash["ACTION_panup"])
    self.panMenu.addAction(actionHash["ACTION_pandown"])
    self.viewMenu.addSeparator()
    self.viewMenu.addAction(actionHash["ACTION_day"])
    self.viewMenu.addAction(actionHash["ACTION_night"])
    self.viewMenu.addSeparator()

    self.viewMenu.setTearOffEnabled(True)
    self.zoomMenu.setTearOffEnabled(True)
    self.panMenu.setTearOffEnabled(True)

def createSettingsMenu(self):  # void MainWindow::createSettingsMenu()
    """
    TOWRITE
    """
    actionHash = self.actionHash
    qDebug("MainWindow createSettingsMenu()")
    self.menuBar().addMenu(self.settingsMenu)
    self.settingsMenu.addAction(actionHash["ACTION_settingsdialog"])
    self.settingsMenu.addSeparator()
    self.settingsMenu.setTearOffEnabled(True)

def createWindowMenu(self):  #void MainWindow::createWindowMenu()
    """
    TOWRITE
    """
    qDebug("MainWindow createWindowMenu()")
    self.menuBar().addMenu(self.windowMenu)
    self.connect(self.windowMenu, SIGNAL("aboutToShow()"), self, SLOT("windowMenuAboutToShow()"))
    # Do not allow the Window Menu to be torn off. It's a pain in the ass to maintain.
    self.windowMenu.setTearOffEnabled(False)

def createHelpMenu(self):  # void MainWindow::createHelpMenu()
    """
    TOWRITE
    """
    actionHash = self.actionHash
    qDebug("MainWindow createHelpMenu()")
    self.menuBar().addMenu(self.helpMenu)
    self.helpMenu.addAction(actionHash["ACTION_help"])
    self.helpMenu.addSeparator()
    self.helpMenu.addAction(actionHash["ACTION_changelog"])
    self.helpMenu.addSeparator()
    self.helpMenu.addAction(actionHash["ACTION_tipoftheday"])
    self.helpMenu.addSeparator()
    self.helpMenu.addAction(actionHash["ACTION_about"])
    self.helpMenu.addSeparator()
    self.helpMenu.addAction(actionHash["ACTION_whatsthis"])
    self.helpMenu.setTearOffEnabled(True)

def createAllMenus(self):  # void MainWindow::createAllMenus()
    """
    TOWRITE
    """
    qDebug("MainWindow createAllMenus()")
    self.createFileMenu()
    self.createEditMenu()
    self.createViewMenu()
    self.createSettingsMenu()
    self.createWindowMenu()
    self.createHelpMenu()


# kate: bom off; indent-mode python; indent-width 4; replace-trailing-space-save on;
